### PR TITLE
Release 1.82.4

### PR DESCRIPTION
**Pull request by harishxr**
### Testing

New tests cover the changes: 

### Description for the changelog

**Does this PR include breaking model changes? If so, Have you added transformation functions?**
  

### Licensing

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,4 +1,14 @@
 # Changelog
+# 1.82.4
+* Enhancement - Add retries to DockerClient's manifest pull functionality and add utils/reference package [#4150](https://github.com/aws/amazon-ecs-agent/pull/4150)
+* Enhancement - ContainerRestartPolicy ACS and TCS models, TMDS updates [#4158](https://github.com/aws/amazon-ecs-agent/pull/4158)[#4163](https://github.com/aws/amazon-ecs-agent/pull/4163)
+* Enhancement - Add restart tracker and restart policy to ecs-agent module
```
Comment (singholt) [inline CHANGELOG.md:5]: missing PR link

Comment (harishxr) [reply CHANGELOG.md:5]: PR not available

Comment (singholt) [reply CHANGELOG.md:5]: https://github.com/aws/amazon-ecs-agent/pull/4162 ?

**Review by singholt — APPROVED** [review]

**Review by yinyic — DISMISSED** [review]

File: agent/version/version.go [hunk 2]
```diff
@@ -22,10 +22,10 @@ package version
...
 // Version is the version of the Agent
-const Version = "1.82.3"
+const Version = "1.82.4"
 
 // GitDirty indicates the cleanliness of the git repo when this agent was built
 const GitDirty = true
 
 // GitShortHash is the short hash of this agent build
-const GitShortHash = "99c5e65a"
+const GitShortHash = "53e8e5ec"
```
Comment (mye956) [inline agent/version/version.go:31]: We might need to update the short sha to be  the one for this [commit](https://github.com/aws/amazon-ecs-agent/commit/428c2f40602927902ff04fbdb3d69dc968466a45) `428c2f40`.  Perhaps try rebasing and re-running the generation scripts again?

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,4 +1,14 @@
...
+# 1.82.4
+* Enhancement - Add retries to DockerClient's manifest pull functionality and add utils/reference package [#4150](https://github.com/aws/amazon-ecs-agent/pull/4150)
+* Enhancement - ContainerRestartPolicy ACS and TCS models, TMDS updates [#4158](https://github.com/aws/amazon-ecs-agent/pull/4158)[#4163](https://github.com/aws/amazon-ecs-agent/pull/4163)
+* Enhancement - Add restart tracker and restart policy to ecs-agent module [#4162](https://github.com/aws/amazon-ecs-agent/pull/4162)
+* Enhancement - Add BackendStatusString method to ContainerStatus [#4167](https://github.com/aws/amazon-ecs-agent/pull/4167)
+* Enhancement - Simplify ContainerStatus.BackendStatusString() method [#4168](https://github.com/aws/amazon-ecs-agent/pull/4168)
+* Enhancement - Bump github.com/docker/docker from 23.0.8+incompatible to 24.0.9+incompatible in /ecs-init [#4122](https://github.com/aws/amazon-ecs-agent/pull/4122)
+* Bugfix - Update ca-certs base image [#4153](https://github.com/aws/amazon-ecs-agent/pull/4153)
+* Bugfix - fixing command typo [#4165](https://github.com/aws/amazon-ecs-agent/pull/4165)
+
```
Comment (mye956) [inline CHANGELOG.md:11]: nit (non-blocking): extra new line

Comment (harishxr) [reply CHANGELOG.md:11]: Resolved

File: agent/version/version.go (hunk 2 above)
Comment (harishxr) [reply agent/version/version.go]: Set short hash to `5847274a`

**Review by mye956 — APPROVED** [review]

**Review by yinyic — APPROVED** [review]

**Pull request merged**